# Módulo 10 tarefa 2

Considere a mesma base de dados utilizada na tarefa 1 ```previsao_de_renda.csv```.

### Importando as Bibliotecas

In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

### Carregando a base de dados "previsao_de_renda.csv'

In [2]:
# importando e visualizando os dados do DataFrame
df = pd.read_csv('previsao_de_renda.csv')
df.head()

Unnamed: 0    data_ref  index sexo  posse_de_veiculo  posse_de_imovel  \
0           0  2015-01-01   8682    F             False             True   
1           1  2015-01-01  12830    M              True             True   
2           2  2015-01-01   2885    M              True             True   
3           3  2015-01-01  16168    F              True            False   
4           4  2015-01-01  12438    M             False            False   

   qtd_filhos   tipo_renda           educacao estado_civil tipo_residencia  \
0           1  Assalariado         Secundário       Casado            Casa   
1           0   Empresário         Secundário       Casado            Casa   
2           2  Assalariado  Superior completo       Casado            Casa   
3           0   Empresário         Secundário       Casado            Casa   
4           0  Assalariado         Secundário       Casado            Casa   

   idade  tempo_emprego  qt_pessoas_residencia    mau    renda  
0     36       3.575342                    3.0  False  3369.24  
1     42       0.860274                    2.0  False  6096.14  
2     31       8.065753                    4.0  False  5658.98  
3     50       1.208219                    2.0  False  7246.69  
4     52      13.873973                    2.0  False  4017.37

In [3]:
# Limpando os dados do DataFrame
df.drop(columns=['Unnamed: 0'], inplace=True)

In [4]:
# Selecionar apenas as variáveis quantitativas do DataFrame
df_quantitative = df.select_dtypes(include=['int64', 'float64'])

In [7]:
df.corr()

<ipython-input-7-2f6f6606aa2c>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


index  posse_de_veiculo  posse_de_imovel  \
index                  1.000000         -0.007542        -0.099683   
posse_de_veiculo      -0.007542          1.000000        -0.014727   
posse_de_imovel       -0.099683         -0.014727         1.000000   
qtd_filhos             0.039734          0.107959        -0.002525   
idade                 -0.047971         -0.157837         0.122667   
tempo_emprego         -0.034574         -0.087514         0.000112   
qt_pessoas_residencia  0.041818          0.156009        -0.007180   
mau                    0.130691         -0.017617        -0.022499   
renda                 -0.001740          0.063414         0.029827   

                       qtd_filhos     idade  tempo_emprego  \
index                    0.039734 -0.047971      -0.034574   
posse_de_veiculo         0.107959 -0.157837      -0.087514   
posse_de_imovel         -0.002525  0.122667       0.000112   
qtd_filhos               1.000000 -0.364303      -0.074250   
idade                   -0.364303  1.000000       0.323934   
tempo_emprego           -0.074250  0.323934       1.000000   
qt_pessoas_residencia    0.896488 -0.338805      -0.065748   
mau                      0.009544 -0.011087      -0.047722   
renda                    0.009567  0.042570       0.394235   

                       qt_pessoas_residencia       mau     renda  
index                               0.041818  0.130691 -0.001740  
posse_de_veiculo                    0.156009 -0.017617  0.063414  
posse_de_imovel                    -0.007180 -0.022499  0.029827  
qtd_filhos                          0.896488  0.009544  0.009567  
idade                              -0.338805 -0.011087  0.042570  
tempo_emprego                      -0.065748 -0.047722  0.394235  
qt_pessoas_residencia               1.000000 -0.001555  0.020977  
mau                                -0.001555  1.000000 -0.011577  
renda                               0.020977 -0.011577  1.000000

1) Com o que você viu em aula, faça uma análise das variáveis qualitativas (posse de imóvel e posse de veículo) para verificar se são boas preditoras de *renda*. Alguma delas parece discriminar *renda* de forma estatisticamente significante?

In [9]:
# Dividir o DataFrame em grupos com base na posse de imóvel
grupo_posse_imovel = df.groupby('posse_de_imovel')['renda'].apply(list)

# Visualizando os dados
grupo_posse_imovel

posse_de_imovel
False    [7246.69, 4017.37, 5577.82, 1728.33, 630.3, 38...
True     [3369.24, 6096.14, 5658.98, 5483.58, 1947.99, ...
Name: renda, dtype: object

In [14]:
# Teste t para comparar as médias de renda entre os grupos
t_stat_imovel, p_valor_imovel = stats.ttest_ind(grupo_posse_imovel[0], grupo_posse_imovel[1])

# Visualizando os dados de teste
print("Posse de imóvel:")
print("   Teste t-valor:", t_stat_imovel)
print("   Valor p:", p_valor_imovel)

Posse de imóvel:
   Teste t-valor: -3.6544544037866884
   Valor p: 0.0002586087928865693


In [12]:
# Dividir o DataFrame em grupos com base na posse de veículo
grupo_posse_veiculo = df.groupby('posse_de_veiculo')['renda'].apply(list)

# visualizando os grupo
grupo_posse_veiculo

posse_de_veiculo
False    [3369.24, 4017.37, 5483.58, 1947.99, 5577.82, ...
True     [6096.14, 5658.98, 7246.69, 7781.51, 2779.67, ...
Name: renda, dtype: object

In [15]:
# Teste t para comparar as médias de renda entre os grupos
t_stat_veiculo, p_valor_veiculo = stats.ttest_ind(grupo_posse_veiculo[0], grupo_posse_veiculo[1])

# Visualizando o valores de teste
print("Posse de veículo:")
print("   Teste t-valor:", t_stat_veiculo)
print("   Valor p:", p_valor_veiculo)

Posse de veículo:
   Teste t-valor: -7.781733886565539
   Valor p: 7.618601898603797e-15


## Insights

Com base nos resultados dos testes t para as variáveis qualitativas "posse de imóvel" e "posse de veículo", podemos extrair os seguintes insights:

### Posse de Imóvel:

* O teste t-valor é -3.65.
*O valor p é 0.00026.
*O valor p muito baixo (menor que 0.05) sugere que há uma diferença estatisticamente significativa nas médias de renda entre os grupos definidos pela posse de imóvel.
*Isso indica que a posse de imóvel pode ser um fator discriminante importante na determinação da renda dos indivíduos no conjunto de dados. Pessoas com posse de imóvel tendem a ter uma renda significativamente diferente daquelas sem posse de imóvel.


### Posse de Veículo:

* O teste t-valor é -7.78.
* O valor p é 7.62e-15.
* O valor p extremamente baixo (praticamente zero) indica uma diferença estatisticamente significativa nas médias de renda entre os grupos definidos pela posse de veículo.
* Isso sugere que a posse de veículo é um preditor muito forte da renda no conjunto de dados. Indivíduos com posse de veículo têm uma renda

### consideravelmente diferente daqueles sem posse de veículo.


Esses insights sugerem que tanto a posse de imóvel quanto a posse de veículo são variáveis importantes na predição da renda das pessoas no conjunto de dados. Essas descobertas podem ter implicações significativas em análises mais detalhadas ou na construção de modelos preditivos de renda.
